### ISSUE TITLE
Generate random set of "n" states of dimension "d"

### ISSUE THREAD

**Issue by vprusso** [vprusso/toqito#380]
**Task**: Generate a collection of "n" random states each of dimension "d".

```py
from qiskit.quantum_info.states.random import random_statevector

def random_states(n: int, d: int) -> list[np.ndarray]:
    """Construct a set of `n` random states of dimension `d`."""
    return [random_statevector(d).data.reshape(-1, 1) for _ in range(n)]  
```

Note that this would also require us to pin `qiskit` as a dependency.

**Comment by ankit-pn** [comment]
I want to work on this issue, it would be great if anyone assign this to me!

**Comment by vprusso** [comment]
Hi @ankit-pn. Thank you for your interest in `toqito` and this issue. 

If you are interested in this, let me provide a bit more implementation-dependent information that may be helpful.

1. The `random_states` function should be placed in `toqito/rand/random_states.py`. The function should also have a docstring that is more complete than the one that it has now (i.e. it should be formatted in a similar style to other functions within `toqito`). In short, it should contain a description, an example, and specify the parameters and returns in the docstring.
2. There should be a set of tests in `rand/tests/test_random_states.py`. Looking at the other `test_*` prefixed files can serve as a guide for how to implement.
3. Once the above is implemented, you'll need to add the line `toqito.rand.random_states` to `docs/rand.rst`.
4. Assuming the linter and tests pass with full coverage, that should be good to go!

Let me know if you have any questions, and thank you again for your interest and contribution, @ankit-pn

**Comment by ankit-pn** [comment]
rand_states.py and test_rand_states.py doesn't exist, i should create them right?

**Comment by ankit-pn** [comment]
The function [ rand_states ] that you have mentioned in this issue is correct ?
I should include this function alongside with its test according to instructions that you have given about?

```python

from qiskit.quantum_info.states.random import random_statevector
def random_states(n: int, d: int) -> list[np.ndarray]:
"""Construct a set of `n` random states of dimension `d`."""
return [random_statevector(d).data.reshape(-1, 1) for _ in range(n)]

```

**Comment by vprusso** [comment]
> rand_states.py and test_rand_states.py doesn't exist, i should create them right?

Yes, this would be part of the task. And I would suggest `random_states.py` and `test_random_states.py` as names instead to match the function name. 

> I should include this function alongside with its test according to instructions that you have given about?

Yes, that's correct. Basically, following the steps outlined in the above comment is what you would need to do.

**Comment by purva-thakre** [comment]
Fixed by #383
